app/scripts/ui/ZipcodeInput.js

```diff
@@ -6,16 +6,33 @@ var Button  = require('react-bootstrap/Button');
 
 var ZipcodeInput = React.createClass({
     displayName: 'ZipcodeInput',
+    validate: function(e){
+        var zipcodeRegex = /^\d{6}$/;
+        var zipcode = e.target.value;
+        if( zipcode.length ){
+            if( zipcodeRegex.test(zipcode) ){
+                $('.goButton').removeAttr('disabled');
```

Comment:
We should do it in the React way. Directly manipulating the DOM is not ideal, as we cannot unit test the validate function at all. Besides, if we directly change the DOM, we'll lose the benefits of the virtual DOM rendering offered by ReactJS.

The right way is to change the state of the component when the validation fails. The view should display the validation message based on this state. 

use `setState()` methods provided by React